mod-data-import-converter-storage-server/src/test/java/org/folio/rest/impl/MappingProfileTest.java

```diff
@@ -761,6 +764,75 @@ public void shouldUpdateProfileOnPutAndReplaceExistingAssociationWithActionProfi
       .body("profileAssociations[0].detailProfileId", is(mappingProfile2.getProfile().getId()));
   }
 
+
+  @Test
+  public void shouldNotCreateMappingProfilesWhenDifferentFolioRecord() {
+    var actionProfileUpdateDto = postActionProfile(new ActionProfileUpdateDto()
+      .withProfile(new ActionProfile()
+        .withName("Test Action Profile")
+        .withAction(CREATE)
+        .withFolioRecord(MARC_BIBLIOGRAPHIC)));
+
+    RestAssured.given()
+      .spec(spec)
+      .body(new MappingProfileUpdateDto()
+        .withProfile(new MappingProfile()
+          .withName("Test Action Profile")
+          .withExistingRecordType(EntityType.INSTANCE)
+          .withIncomingRecordType(EntityType.MARC_BIBLIOGRAPHIC)
+          .withParentProfiles(List.of(new ProfileSnapshotWrapper()
+            .withId(actionProfileUpdateDto.getProfile().getId())
+            .withContentType(ACTION_PROFILE)
+            .withContent(actionProfileUpdateDto)))))
```

Comment:
please, use withAddedRelations() method on MappingProfileUpdateDto to link action profile to the mapping profile, like it was done in the test for action profile.